pkg/apis/pipeline/v1/result_types.go

```diff
@@ -38,6 +38,27 @@ type TaskResult struct {
 	Value *ResultValue `json:"value,omitempty"`
 }
 
+// StepResult used to describe the Results of a Step.
+//
+// This is field is at an ALPHA stability level and gated by "enable-step-actions" feature flag.
+type StepResult struct {
+	// Name the given name
+	Name string `json:"name"`
+
+	// Type is the user-specified type of the result. The possible type
+	// is currently "string" and will support "array" in following work.
```

Comment:
I think we support array and object? Just realized that the task result type docstrings also need to be updated
```suggestion
	// The possible types are 'string', 'array', and 'object', with 'string' as the default.
```

Reply:
done! Thanks